### ISSUE TITLE
Request: preferring `requirements.in` over `requirements.txt`

### ISSUE THREAD

**Issue by jamesbraza** [arkq/flake8-requirements#84]
Those using [pip-tools](https://github.com/jazzband/pip-tools) or [uv](https://github.com/astral-sh/uv) will have a `requirements.in` that is compiled to a `requirements.txt`. That means:
- Known dependencies: entered into `requirements.in`
- Known dependencies + random indirect dependencies: compiled in `requirements.txt`

`flake8-requirements` should check the `requirements.in` file, so requirements are actually ensured to be present (and not a byproduct of compilation pulling in indirect dependencies).

In other words, it would be valuable for the requirements inference (in [`get_mods_3rd_party_requirements`](https://github.com/arkq/flake8-requirements/blob/v2.0.2/src/flake8_requirements/checker.py#L702)) to prefer a `requirements.in` file over a `requirements.txt` file, when a `requirements.in` file is present.  I supposed there should also be an opt-out argument `--ignore-in-file` to turn off looking for a `requirements.in`.

**Comment by arkq** [comment]
How about using already existing mechanizm of specifying requirements file manually, e.g:
```
flake8 --requirements-file requirements.in
```

**Comment by jamesbraza** [comment]
Yeah that can work too. Though I think it'd be cool if things worked automagically without requiring arg overrides 😄 

Feel free to close this out if it's not something `flake8-requirements` wants to support.

**Comment by arkq** [comment]
I'd like to keep additional default files to scan to minimum. Since your use case is possible with currently available features I'm closing this issue.
